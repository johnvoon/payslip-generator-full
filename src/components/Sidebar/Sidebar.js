import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router";

export default class Sidebar extends Component {
  renderLinks() {
    const { links, handleHide } = this.props;

    return links.map((link, idx) => {
      if (link) {
        const linkText = link[0];
        const slug = link[1];

        return (
          <Link
            key={idx}
            className="list-group-item"
            to={slug}
            onClick={handleHide}
          >
            {linkText}
          </Link>
        );
      } else return null;
    });
  }

  render() {
    return (
      <div className="sidebar pull-right">
        <div className="list-group">
          <div className="list-group-item">
            <h4 className="list-group-item-heading">
              Menu
            </h4>
          </div>
          {this.renderLinks()}
        </div>
      </div>
    );
  }
}

Sidebar.propTypes = {
  links: PropTypes.array.isRequired,
  handleHide: PropTypes.func.isRequired
};
