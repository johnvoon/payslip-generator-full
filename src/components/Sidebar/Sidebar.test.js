import React from "react";
import Sidebar from "./Sidebar";
import { shallow } from "enzyme";

describe("Sidebar", () => {
  const links = [
    ["Home", "/", "index"],
    ["Payslip Generator", "/payslip-generator"]
  ];
  const handleHide = jest.fn();
  const renderedComponent = shallow(
    <Sidebar links={links} handleHide={handleHide} />
  );

  it("should match snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
