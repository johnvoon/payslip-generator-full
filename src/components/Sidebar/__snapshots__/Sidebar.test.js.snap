exports[`Sidebar should match snapshot 1`] = `
<div
  className="sidebar pull-right">
  <div
    className="list-group">
    <div
      className="list-group-item">
      <h4
        className="list-group-item-heading">
        Menu
      </h4>
    </div>
    <Link
      className="list-group-item"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/">
      Home
    </Link>
    <Link
      className="list-group-item"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/payslip-generator">
      Payslip Generator
    </Link>
  </div>
</div>
`;
